File: app/src/main/java/baidu/com/testlibproject/SimpleAdapter.java
package baidu.com.testlibproject;


import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

public class SimpleAdapter extends BaseAdapter {
    private Context mCxt;
    private String[] mStrArr;

    public SimpleAdapter(Context cxt) {
        mCxt = cxt;
    }

    public SimpleAdapter(Context cxt, String[] strArr) {
        mCxt = cxt;
        mStrArr = strArr;
    }

    public void setStrArr(String[] strArr) {
        mStrArr = strArr;
    }

    @Override
    public int getCount() {
        if (mStrArr != null) {
            return mStrArr.length;
        } else {
            return 0;
        }
    }

    @Override
    public String getItem(int position) {
        if (mStrArr != null) {
            return mStrArr[position];
        } else {
            return null;
        }
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {
        LayoutInflater inflater = LayoutInflater.from(mCxt);
        if (convertView == null) {
            convertView = inflater.inflate(R.layout.main_activity_item, null);
        }
        if (mStrArr != null && position <= mStrArr.length) {
            ((TextView) convertView.findViewById(R.id.item_textview)).setText(mStrArr[position]);
        }
        convertView.setTag(position);
        return convertView;
    }
}

File: app/src/main/java/baidu/com/testlibproject/SimpleAdapter.kt
package baidu.com.testlibproject


import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.TextView

class SimpleAdapter : BaseAdapter {
    private var mCxt: Context? = null
    private var mStrArr: Array<String>? = null

    constructor(cxt: Context) {
        mCxt = cxt
    }

    constructor(cxt: Context, strArr: Array<String>) {
        mCxt = cxt
        mStrArr = strArr
    }

    fun setStrArr(strArr: Array<String>) {
        mStrArr = strArr
    }

    override fun getCount(): Int {
        return if (mStrArr != null) {
            mStrArr!!.size
        } else {
            0
        }
    }

    override fun getItem(position: Int): String? {
        return if (mStrArr != null) {
            mStrArr!![position]
        } else {
            null
        }
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
        var convertView = convertView
        val inflater = LayoutInflater.from(mCxt)
        if (convertView == null) {
            convertView = inflater.inflate(R.layout.main_activity_item, parent, false)
        }
        if (mStrArr != null && position <= mStrArr!!.size) {
            (convertView!!.findViewById<View>(R.id.item_textview) as TextView).text = mStrArr!![position]
        }
        convertView!!.tag = position
        return convertView
    }
}
